MapboxCoreNavigation/NavigationLocationManager.swift

```diff
@@ -17,24 +17,37 @@ open class NavigationLocationManager: CLLocationManager {
     
     var lastKnownLocation: CLLocation?
     
+    /**
+     Indicates whether the device is plugged in or not.
+     */
+    public private(set) var isPluggedIn: Bool = false
+    
+    /**
+     Indicates whether the location manager’s desired accuracy should update
+     when the battery state changes.
+     */
+    public var automaticallyUpdatesDesiredAccuracy = true
+    
     override public init() {
         super.init()
         
-        let always = Bundle.main.locationAlwaysUsageDescription
-        let both = Bundle.main.locationAlwaysAndWhenInUseUsageDescription
-        
-        if always != nil || both != nil {
-            requestAlwaysAuthorization()
-        } else {
-            requestWhenInUseAuthorization()
-        }
+        requestWhenInUseAuthorization()
         
         if #available(iOS 9.0, *) {
             if Bundle.main.backgroundModes.contains("location") {
                 allowsBackgroundLocationUpdates = true
             }
         }
         
-        desiredAccuracy = kCLLocationAccuracyBestForNavigation
+        desiredAccuracy = kCLLocationAccuracyBest
+        
+        #if os(iOS)
+            guard automaticallyUpdatesDesiredAccuracy else { return }
+            let _ = UIDevice.current.observe(\.batteryState) { [weak self] (device, changed) in
```

Comment:
Is it necessary to remove this observation when deiniting?

Reply:
That should be handled automatically when the NavigationLocationManager deinits but there are a couple of radars open indicating this doesn't work properly on iOS 10 and below https://openradar.appspot.com/35027212. I think we should hold onto the return value and invalidate it properly. 😞 
Would you mind invalidating the observer in `NavigationMapView` while you're at it?

Reply:
@bsudekum The new KVO API (`someInstance.observe(_ keyPath:closure:)`) does not require removing observers on deinit. That's one of the nice things about it.

Reply:
I wasn't aware of the radar though. 🤔

And it looks like manually removing an observer is even more of a PITA with the new API, maybe we can still call `NotificationCenter.removeObserver(self)`?

Reply:
@JThramer I think @frederoni is saying there could be issues with iOS 10 not properly invalidating the property.